internal/services/postgres/postgresql_flexible_server_virtual_endpoint_resource.go

```diff
@@ -231,17 +242,23 @@ func (r PostgresqlFlexibleServerVirtualEndpointResource) Update() sdk.ResourceFu
 	}
 }
 
-// exposed so we can access from tests
-func DeletePostgresFlexibileServerVirtualEndpoint(ctx context.Context, client *virtualendpoints.VirtualEndpointsClient, id *virtualendpoints.VirtualEndpointId) error {
-	deletePoller := custompollers.NewPostgresFlexibleServerVirtualEndpointDeletePoller(client, *id)
-	poller := pollers.NewPoller(deletePoller, 5*time.Second, pollers.DefaultNumberOfDroppedConnectionsToAllow)
-
-	if _, err := client.Delete(ctx, *id); err != nil {
-		return fmt.Errorf("deleting %s: %+v", *id, err)
+// The flexible endpoint API does not store the location/rg information on replicas it only stores the name.
+// This lookup is safe because replicas for a given source server are *not* allowed to have identical names
```

Comment:
The names may not be unique per subscription, however they will be unique per source server. Since we check that the name matches, as well as filter those results based on the source server ID we should be protected from selecting an incorrect replica. Even if the names matched, replicas must have unique names per source servers.

https://github.com/hashicorp/terraform-provider-azurerm/pull/27509/files/94e28c5de14955a188f90192c06828946f9a6fcc#diff-6fceca959b7c5669ebd391456b3a69c7ff858802d5dc5e70a91c6237cd173ff0R258

I was worried about this as well, but I believe it is setup in a safe manner.